token/core/zkatdlog/crypto/audit/auditor.go

```diff
@@ -262,6 +317,9 @@ func getAuditInfoForTransfers(des Deserializer, transfers [][]byte, metadata []d
 		for i := 0; i < len(tr.SenderAuditInfos); i++ {
 			var matcher driver.Matcher
 			var err error
+			if inputs[k][i] == nil {
+				return nil, nil, errors.Errorf("input[%d][%d] is nil", k, i)
+			}
 			if !inputs[k][i].IsRedeem() {
 				matcher, err = des.GetOwnerMatcher(tr.SenderAuditInfos[i])
 				if err != nil {
```

Comment:
shall we open a github issue to return always a message with the error?

Reply:
I will address it in this PR.